pkg/operator/ceph/config/defaults.go

```diff
@@ -71,33 +74,29 @@ func DefaultCentralizedConfigs(cephVersion version.CephVersion) *Config {
 	if !cephVersion.IsAtLeast(version.CephVersion{Major: 14, Minor: 2, Extra: 1}) {
 		// Set the default log files to empty so they don't bloat containers. Can be changed in
 		// Mimic+ by users if needed.
-		c.Section("global").
-			Set("log file", "").
-			Set("mon cluster log file", "")
+		overrides = append(overrides, []rookceph.ConfigOverride{
+			configOverride("global", "log file", ""),
+			configOverride("global", "mon cluster log file", ""),
+		}...)
 	}
 
-	return c
+	return overrides
 }
 
 // DefaultLegacyConfigs need to be added to the Ceph config file until the integration tests can be
 // made to override these options for the Ceph clusters it creates.
-func DefaultLegacyConfigs() *Config {
-	c := NewConfig()
-	c.Section("global").
-		Set("mon max pg per osd", "1000").
+func DefaultLegacyConfigs() rookceph.ConfigOverridesSpec {
+	overrides := []rookceph.ConfigOverride{
+		configOverride("global", "mon max pg per osd", "1000"),
 		//
-		Set("osd pg bits", "11").
-		Set("osd pgp bits", "11").
-		Set("osd pool default size", "1").
-		Set("osd pool default min size", "1").
-		Set("osd pool default pg num", "100").
-		Set("osd pool default pgp num", "100").
+		// TODO: remove these; if we need for integration tests, set in integration test spec
+		configOverride("global", "osd pool default size", "1"),
+		configOverride("global", "osd pool default min size", "1"),
+		configOverride("global", "osd pool default pg num", "100"),
+		configOverride("global", "osd pool default pgp num", "100"),
 		//
-		Set("rbd_default_features", "3"). // TODO: still needed?
-		// Setting fatal signal handlers to true (the default) will print a lot of extra information
-		// from daemons when they encounter a failures, but it is VERY verbose. When the mon kv
-		// store is available, it will probably be best to set this to false by default if it is
-		// unset, but leave it set to true if the user has specified it as true during runtime.
-		Set("fatal signal handlers", "false")
-	return c
+		// TODO: Is this still necessary?
+		configOverride("global", "rbd_default_features", "3"),
```

Comment:
It's only needed by the client who will create rbd block devices. At this point, Ceph CSI is responsible for that. Instances running on Flex will need that I suppose. So once we drop Flex I'd say, this can be removed. Also, keeping Ceph's default is probably better.